### PR TITLE
linuxPackages.nvidia_x11_legacy390: mark as broken

### PR DESCRIPTION
**Pull request by Kiskae**
###### Description of changes

As of linux 5.17 the removal of the `PDE_DATA()` macro breaks this version of the driver.

###### Things done

Marked it as broken for kernel 5.17 and higher.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -67,6 +67,8 @@ rec {
     sha256_64bit = "09qcdfn4j5jza3iw59wqwgq4a489qf7kx355yssrcahaw9g87lxz";
     settingsSha256 = "16qqw0jy31da65cdi17y3j2kcdhw09vra7g17bkcimaqnf70j0ni";
     persistencedSha256 = "1ad81y4qfpxrx0vqsk81a3h0bi1yg8hw5gi5y5d58p76vc8083i9";
+
+    broken = with kernel; kernelAtLeast "5.17";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:71]:
```suggestion
    broken = kernel.kernelAtLeast "5.17";
```

Comment (Kiskae) [reply pkgs/os-specific/linux/nvidia-x11/default.nix]: fixed this as well as the same line for `legacy340`, then squashed it to a single commit.

**Comment by Kiskae** [comment]
I'd like to add that the "new feature branch" build listed on https://www.nvidia.com/en-us/drivers/unix/ (`495.46`) also has the same issue. I guess if the production branch is newer it supersedes it.

**Comment by wegank** [comment]
Also #166410. An eventual patch would simply be like [replacing](https://github.com/torvalds/linux/commit/359745d78351c6f5442435f81549f0207ece28aa) `PDE_DATA()` with `pde_data()`.

**Pull request merged**